lib/Runtime/Library/InJavascript/Intl.js

```diff
@@ -954,18 +955,23 @@
 
             const thisStr = String(this);
             const thatStr = String(that);
-            const stateObject = _.create();
-            InitializeCollator(stateObject, locales, options);
-
-            return Number(platform.compareString(
-                thisStr,
-                thatStr,
-                stateObject.locale,
-                stateObject.sensitivityEnum,
-                stateObject.ignorePunctuation,
-                stateObject.numeric,
-                stateObject.caseFirstEnum
-            ));
+
+            // Performance optimization to cache the state object and UCollator when the default arguments are provided
```

Comment:
>default arguments [](start = 89, length = 17)

@jackhorton I thought the other caches were already more sophisticated than this -- maybe I'm just remembering our discussion on caching "nice-to-have's"

Reply:
This is caching the Intl.Collator object for `"a".localeCompare("b")`. The PR also caches the UCollator object for each Intl.Collator. This brings Collator in line with how NumberFormat and DateTimeFormat work (the actual implementation is more like DateTimeFormat, where we lazy-initialize the underlying ICU object on the first call to `compare()`).

Reply:
Gotcha -- this block is more about the lazy init and caching of the underlying object? But in addition don't we have a multiple-entry-deep cache for previously constructed Intl objects that works for more than the default params in general? (I guess that wasn't touched by this PR.)

Reply:
We have that DateToLocaleStringCache or something since there are multiple toLocaleStrings and they all want the same caching behavior, this block accomplishes the same thing. The lazy-initialization and caching of the UCollator happens in IntlBlahBlahBlah.cpp